match/lib/match/encryption/openssl.rb

```diff
@@ -0,0 +1,149 @@
+require 'base64'
+require 'openssl'
+require 'securerandom'
+require 'security'
+require 'shellwords'
+
+require_relative '../module'
+require_relative '../change_password'
+
+module Match
+  module Encryption
+    class OpenSSL < Interface
+      attr_accessor :git_url
+
+      attr_accessor :working_directory
+
+      # @param git_url: The Git URL is used for identifiying a specific repo
+      #                 which is used to store the passphrase in the Keychain
+      # @param working_directory: The path to where the certificates are stored
+      def initialize(git_url: nil, working_directory: nil)
```

Comment:
I _think_ we should actually rename this from `git_url` to `server_name` or `keychain_key` since this should be storage agnostic 🙃